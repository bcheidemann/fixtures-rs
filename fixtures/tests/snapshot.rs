use indoc::indoc;
use std::process::Command;

fn test_example_expansion(example_name: &str) {
    let output = Command::new("cargo")
        .args(["expand", "--tests", "--example", example_name])
        .output()
        .expect("failed to expand example");

    if !output.status.success() {
        let stdout = String::from_utf8_lossy(&output.stdout);
        let stderr = String::from_utf8_lossy(&output.stderr);
        panic!(
            indoc! {"
                error: failed to expand example ({})

                -- begin stdout --
                {}
                -- end stdout --

                -- begin stderr --
                {}
                -- end stderr --
            "},
            example_name, stdout, stderr
        );
    }

    let current_dir = std::env::current_dir()
        .expect("failed to get current directory")
        .to_string_lossy()
        .to_string();
    let expansion = String::from_utf8_lossy(&output.stdout).replace(&current_dir, "<repo>");

    insta::assert_snapshot!(example_name, expansion);
}

#[test]
fn basic_usage() {
    test_example_expansion("basic_usage");
}

#[test]
fn complex_signatures() {
    test_example_expansion("complex_signatures");
}

#[test]
fn criterion() {
    test_example_expansion("criterion");
}

#[test]
fn ignore_attributes() {
    test_example_expansion("ignore_attributes");
}

#[test]
fn multiple_fixtures() {
    test_example_expansion("multiple_fixtures");
}

#[test]
fn negative_globs() {
    test_example_expansion("negative_globs");
}
