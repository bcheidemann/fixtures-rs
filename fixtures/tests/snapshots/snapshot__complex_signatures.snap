---
source: fixtures/tests/snapshot.rs
expression: expansion
---
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use std::path::Path;
use fixtures::fixtures;
fn complex_fn(
    _path: &Path,
    _owned: String,
    _mutable: &mut String,
    _borrowed: &str,
) -> Result<String, ()> {
    Ok("hello world!".to_string())
}
fn complex_fn_file_1_dot_txt(
    _owned: String,
    _mutable: &mut String,
    _borrowed: &str,
) -> Result<String, ()> {
    complex_fn::file_1_dot_txt(_owned, _mutable, _borrowed)
}
mod complex_fn {
    use super::*;
    pub fn file_1_dot_txt(
        _owned: String,
        _mutable: &mut String,
        _borrowed: &str,
    ) -> Result<String, ()> {
        complex_fn(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/complex_signatures/file_1.txt",
            ),
            _owned,
            _mutable,
            _borrowed,
        )
    }
    pub const EXPANSIONS: &[fn(
        _owned: String,
        _mutable: &mut String,
        _borrowed: &str,
    ) -> Result<String, ()>] = &[file_1_dot_txt];
}
#[allow(dead_code)]
fn main() {
    complex_fn::file_1_dot_txt(
            "owned".to_string(),
            &mut "mutable".to_string(),
            "borrowed",
        )
        .unwrap();
}
#[rustc_main]
#[coverage(off)]
#[doc(hidden)]
pub fn main() -> () {
    extern crate test;
    test::test_main_static(&[])
}
